core/src/main/java/io/grpc/internal/ClientStream.java

```diff
@@ -72,6 +73,7 @@
    * <p>This method should not throw any exceptions.
    *
    * @param listener non-{@code null} listener of stream events
+   * @param headers to send at the beginning of the call
    */
-  void start(ClientStreamListener listener);
+  void start(ClientStreamListener listener, Metadata headers);
```

Comment:
Done. Removed Metadata from newStream

The non trivial changes are in:
InProcessTransport
DelayedStream
MetadataApplierImpl

I added a new setStream API to DelayedStream which taks a Metadata param. 
I also made MetadataApplierImpl to create a DelayedStream in all cases. Because it seems easier and cleaner. Alternatively, if a faster path is desired, I think I could add a delegate stream class to apply the Metadata for non-delayed streams.

PTAL. Thanks.